AWS: add ACL support for S3FileIO

jackye1995 opened the pull request:
As discussed in #1767 with @danielcweeks and @johnclara, add ACL support for `S3FileIO`

johnclara commented:
lgtm!

danielcweeks commented:
@jackye1995 and @johnclara we should probably figure out #1786 and make the ACL configuration follow a similar pattern.  It looks like we need to rebase on top of #1767 and add the configuration to the `CreateMultipartUploadRequest` as well.

jackye1995 commented:
> @jackye1995 and @johnclara we should probably figure out #1786 and make the ACL configuration follow a similar pattern. It looks like we need to rebase on top of #1767 and add the configuration to the `CreateMultipartUploadRequest` as well.

updated, following the same pattern as #1786 . I am intentionally separating them into `configureEncryption` and `configurePermission` to be explicit about the intention of the methods.

danielcweeks commented on aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java:
```diff
@@ -165,6 +177,11 @@ public AwsProperties(Map<String, String> properties) {
+    String aclType = properties.get(S3FILEIO_ACL);
+    this.s3FileIoAcl = ObjectCannedACL.fromValue(aclType);
+    Preconditions.checkArgument(s3FileIoAcl == null || !s3FileIoAcl.equals(ObjectCannedACL.UNKNOWN_TO_SDK_VERSION),
+        "Cannot supported S3 CannedACL " + aclType);
```
nit: did you mean "Unsupported S3 CannedACL"?

danielcweeks commented:
@jackye1995 One very minor nit, but other than that, this looks good to me.  If you could just update the message, I'll push it.

jackye1995 commented:
> @jackye1995 One very minor nit, but other than that, this looks good to me. If you could just update the message, I'll push it.

Thank you! Just fixed #1786 , I think there will be some merge conflicts with this one, if you can first merge that PR, I can rebase this one with the fix to the error message.

danielcweeks commented:
> > @jackye1995 One very minor nit, but other than that, this looks good to me. If you could just update the message, I'll push it.
> 
> Thank you! Just fixed #1786 , I think there will be some merge conflicts with this one, if you can first merge that PR, I can rebase this one with the fix to the error message.

Both look good.  Just waiting on checks for #1786 and I'll push that one.

jackye1995 commented:
> > > @jackye1995 One very minor nit, but other than that, this looks good to me. If you could just update the message, I'll push it.
> > 
> > 
> > Thank you! Just fixed #1786 , I think there will be some merge conflicts with this one, if you can first merge that PR, I can rebase this one with the fix to the error message.
> 
> Both look good. Just waiting on checks for #1786 and I'll push that one.

fixed
